boot/systems.go

```diff
@@ -58,14 +58,24 @@ func ClearTryRecoverySystem(dev Device, systemLabel string) error {
 		return err
 	}
 
+	modified := false
 	// we may be repeating the cleanup, in which case the system was already
 	// removed from the modeenv and we don't need to rewrite the modeenv
 	if updated, found := dropFromRecoverySystemsList(m.CurrentRecoverySystems, systemLabel); found {
 		m.CurrentRecoverySystems = updated
+		modified = true
+	}
+	if m.TryModel != "" {
+		// recovery system is tried with a matching models
+		m.clearTryModel()
```

Comment:
this will not disrupt in case of reboot later uses for TryModel because we call ClearTryRecoverySystem only if we were at the point to trying one? later uses of TryModel should not arrive here, is that correct?

Reply:
Yes, ClearTryRecoverySystem is only called if after inspecting the bootenv the outcome is that a recovery system was tried. In the usual case of a random reboot, no recovery system will be tried, hence clear will not be called, see https://github.com/snapcore/snapd/blob/master/overlord/devicestate/devicemgr.go#L1150-L1155